### PR TITLE
(WIP) Sync themes

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

* syncs the themes directory in devtools-modules
  * everything is copied over
  * i re-added `@namespace theme-light;` for the launchpad
* launchpad webpack is upgraded
  * supports svg,png, ...
  * maps protocols to devtools-modules/src/themes

#### Why CP the entire themes dir? 

1. having everything encourage's re-use. we want to share our components/themes etc
2. we'll slowly add everything... we're already pulling in a lot of images and files
3. it's easier...

#### What's this @namespace thing

see this PR where we drop the namespace in favor of a new PostCSS plugin. It would be cool to add the `@namespace theme-light` to MC which should be inert because it's not a well used feature.

#### Next Steps

the svgs/pngs are not loading yet, but atleast they're not exploding...

screen shot 2017-09-25 at 9 25 53 pm

### CODE REVIEW

File: packages/devtools-launchpad/.babelrc [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
+    [
+      "module-resolver",
+      {
+        "alias": {
+          "devtools/client/shared/vendor/react": "react",
+          "devtools/client/shared/vendor/react-dom": "react-dom"
+        }
       }
-    }]
+    ]
```
Comment (jasonLaster) [inline packages/devtools-launchpad/.babelrc]: so happy to drop the dist alias

File: packages/devtools-launchpad/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "devtools-launchpad",
-  "version": "0.0.100",
+  "version": "0.0.101-rc4",
```
Comment (jasonLaster) [inline packages/devtools-launchpad/package.json]: these will all go when it's time

File: packages/devtools-launchpad/package.json [hunk 3]
```diff
@@ -18,8 +18,7 @@
     "build-storybook": "build-storybook",
     "license-check": "devtools-license-check",
     "start": "node bin/dev-server",
-    "test": "jest",
-    "prepublish": "node bin/validate-deps"
```
Comment (jasonLaster) [inline packages/devtools-launchpad/package.json]: i should bring this back

File: packages/devtools-launchpad/package.json [hunk 4]
```diff
@@ -32,6 +31,7 @@
     "node": ">=7.0.0"
   },
   "dependencies": {
+    "alias-resolve-loader": "^2.0.4",
```
Comment (jasonLaster) [inline packages/devtools-launchpad/package.json]: i can kill this

File: packages/devtools-launchpad/package.json [hunk 5]
```diff
@@ -59,14 +59,15 @@
...
-    "devtools-modules": "^0.0.31",
+    "devtools-modules": "^0.0.32-rc2",
     "devtools-sprintf-js": "^1.0.3",
     "devtools-utils": "^0.0.5",
     "express": "^4.13.4",
     "extract-text-webpack-plugin": "^3.0.0",
     "fs-extra": "^2.0.0",
     "fuzzaldrin-plus": "^0.4.0",
     "geckodriver": "=1.4.0",
+    "image-webpack-loader": "^3.4.2",
```
Comment (jasonLaster) [inline packages/devtools-launchpad/package.json]: this too

File: packages/devtools-launchpad/package.json [hunk 6]
```diff
@@ -75,16 +76,13 @@
     "postcss": "^6.0.6",
     "postcss-bidirection": "^2.4.0",
     "postcss-class-namespace": "^0.1.0",
+    "postcss-inline-base64": "^2.2.1",
```
Comment (jasonLaster) [inline packages/devtools-launchpad/package.json]: kill

File: packages/devtools-launchpad/package.json [hunk 7]
```diff
@@ -75,16 +76,13 @@
...
     "postcss-loader": "^2.0.6",
+    "postcss-map-url": "^1.0.2",
```
Comment (jasonLaster) [inline packages/devtools-launchpad/package.json]: kill

File: packages/devtools-launchpad/package.json [hunk 8]
```diff
@@ -75,16 +76,13 @@
...
+    "postcss-url-mapper": "^1.0.2",
```
Comment (jasonLaster) [inline packages/devtools-launchpad/package.json]: keep

File: packages/devtools-launchpad/postcss.config.js [hunk 9]
```diff
@@ -1,7 +1,23 @@
...
+function myMapFunc(url) {
+  const urls = [
+    "chrome://devtools/skin/themes",
+    "chrome://devtools/skin",
+    "resource://devtools/client/themes"
+  ];
+  const r = new RegExp(`(${urls.join("|")})`);
+  const newUrl = url.replace(r, "/themes");
+
+  console.log(`PC: ${newUrl} \t\t ${url}`);
```
Comment (jasonLaster) [inline packages/devtools-launchpad/postcss.config.js]: kill

File: packages/devtools-launchpad/webpack.config.js [hunk 10]
```diff
@@ -50,7 +52,7 @@ module.exports = (webpackConfig, envConfig, options = {}) => {
   });
 
   webpackConfig.module.rules.push({
-    test: /\.svg$/,
+    test: /^(?!css).*svg$/,
```
Comment (jasonLaster) [inline packages/devtools-launchpad/webpack.config.js]: i'd like to take this out in a second pass

File: packages/devtools-launchpad/webpack.config.js [hunk 11]
```diff
@@ -81,7 +83,28 @@ module.exports = (webpackConfig, envConfig, options = {}) => {
       test: /\.css$/,
       use: [
         { loader: "style-loader" },
-        { loader: "css-loader", options: { importLoaders: 1 } },
+        // {
+        //   enforce: "pre",
```
Comment (jasonLaster) [inline packages/devtools-launchpad/webpack.config.js]: kill

File: packages/devtools-launchpad/webpack.config.js [hunk 12]
```diff
@@ -109,6 +132,26 @@ module.exports = (webpackConfig, envConfig, options = {}) => {
...
+          "./chrome://devtools/skin/themes",
+          "./chrome://devtools/skin",
+          "./chrome://devtools/content/shared",
+          "./chrome://devtools/content",
+          "./resource://devtools/client/themes"
+        ];
+        const r = new RegExp(`(${urls.join("|")})`);
+
+        const newUrl = resource.request.replace(r, `.`);
+        console.log(`URL: ${newUrl} \t\t ${resource.request}`);
```
Comment (jasonLaster) [inline packages/devtools-launchpad/webpack.config.js]: kill

**Comment by jasonLaster** [comment]
Alright, i think this is ready for review.

The beautiful thing is that this change actually takes all the themes out of devtools-core so that we can keep them in a package that is JUST for syncing. And we pull them in through NPM.

File: README.md [hunk 13]
```diff
@@ -66,3 +72,5 @@
 [devtools-splitter-version]:https://img.shields.io/npm/v/devtools-splitter.svg
 [devtools-splitter-pkg]:https://npmjs.org/package/devtools-splitter
 [devtools-splitter]:./packages/devtools-splitter/#readme
+
+[dma]: https://github.com/jasonLaster/devtools-mc-assets
```
Comment (nchevobbe) [inline README.md]: could this be moved to devtools-html at some point ?

File: packages/devtools-launchpad/postcss.config.js [hunk 14]
```diff
@@ -1,7 +1,22 @@
+var mapUrl = require("postcss-url-mapper");
```
Comment (nchevobbe) [inline packages/devtools-launchpad/postcss.config.js]: should this have the license header ?

File: packages/devtools-launchpad/src/components/LandingPage.js [hunk 15]
```diff
@@ -98,24 +100,39 @@ const LandingPage = React.createClass({
...
-                              isUnderConstruction ? this.renderExperimentalMessage(browserName) // eslint-disable-line max-len
-                                                  : null);
+    return dom.div(
+      { className: "launch-action-container" },
+      dom.button(
+        { onClick: () => this.launchBrowser(browserName) },
+        `Launch ${browserName}`
+      ),
+      isUnderConstruction
+        ? this.renderExperimentalMessage(browserName) // eslint-disable-line max-len
```
Comment (nchevobbe) [inline packages/devtools-launchpad/src/components/LandingPage.js]: remove eslint disable which shouldn't be needed anymore

File: packages/devtools-launchpad/src/components/LandingPage.js [hunk 16]
```diff
@@ -98,24 +100,39 @@ const LandingPage = React.createClass({
...
+    const underConstructionMessage =
+      "Debugging is experimental and certain features won't work (i.e, seeing variables, attaching breakpoints)"; // eslint-disable-line max-len
+    const githubIssuesUrl =
+      "https://github.com/devtools-html/debugger.html/issues?q=is%3Aopen+is%3Aissue+label%3A";
+
+    return dom.div(
+      { className: "under-construction" },
+      dom.div(
+        { className: "under-construction-message" },
+        dom.p({}, underConstructionMessage), // eslint-disable-line max-len
```
Comment (nchevobbe) [inline packages/devtools-launchpad/src/components/LandingPage.js]: remove eslint disable which shouldn't be needed anymore

File: packages/devtools-launchpad/webpack.config.js [hunk 17]
```diff
@@ -50,18 +52,9 @@ module.exports = (webpackConfig, envConfig, options = {}) => {
   });
 
   webpackConfig.module.rules.push({
-    test: /\.svg$/,
```
Comment (nchevobbe) [inline packages/devtools-launchpad/webpack.config.js]: didn't look at the following yet, but if there are no changes in reps, I think that will break them.

Comment (jasonLaster) [reply packages/devtools-launchpad/webpack.config.js]: yep... reps will need to stay on the old launchpad for a bit. same w/ the dbg

**Pull request merged**